### PR TITLE
feat: reflect validationIcon property

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #8057

## Summary

- Reflect the `validationIcon` - brought up in https://github.com/Esri/calcite-design-system/pull/8561#issuecomment-1884074225
- Add `validationIcon`, `validationMessage`, and `status` properties to the reflects and defaults common tests - brought up in https://github.com/Esri/calcite-design-system/pull/8561#issuecomment-1879355882

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/src/components/select/select.e2e.ts [hunk 1]
```diff
@@ -37,6 +38,15 @@ describe("calcite-select", () => {
     focusable(simpleTestMarkup);
   });
 
+  describe("defaults", () => {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/select/select.e2e.ts:41]: ✨🏆✨

**Review by jcfranco — APPROVED** [review]
✅

**Comment by jcfranco** [comment]
@benelan Should this be `feat` since it's enhancing the behavior?

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts [hunk 2]
```diff
@@ -35,6 +36,23 @@ describe("calcite-input-date-picker", () => {
...
+      {
+        propertyName: "status",
+        value: "invalid",
+      },
+      {
+        propertyName: "validationIcon",
+        value: true,
+      },
+    ]);
+  });
```
Comment (benelan) [inline packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts:54]: @jcfranco not sure why the reflects test for all three of these properties are failing, but I'm going to remove them for now

**Pull request merged**